class.jetpack-cli.php

```diff
@@ -934,7 +934,10 @@ public function partner_provision( $args, $named_args ) {
 			: false;
 
 		/** This filter is documented in class.jetpack-cli.php */
-		if ( apply_filters( 'jetpack_start_enable_sso', true ) ) {
+		if (
+			( ! Jetpack::is_active() || Jetpack::is_module_active( 'sso' ) ) &&
```

Comment:
Rather than making this part of the condition, why not make it the default value?

i.e. 

```php
apply_filters( 'jetpack_start_enable_sso', ( ! Jetpack::is_active() || Jetpack::is_module_active( 'sso' ) ) )
```